### ISSUE TITLE
[Bug Report][2.6.2] v-calendar should use table elements in markup

### ISSUE THREAD

**Issue by apfrod** [vuetifyjs/vuetify#14604]
### Environment
**Vuetify Version:** 2.6.2
**Vue Version:** 2.6.14
**Browsers:** Chrome 97.0.4692.71
**OS:** Mac OS 10.15.7

### Steps to reproduce
I have received the following report from an accessibility tester regarding v-calendar:
The table content such as "SUN", "MON", "TUE" etc. is presented using `<div>` elements.
As a result, screen reader users cannot understand the data table content effectively.

### Expected Behavior
Apply the following changes:
- Utilized table mark-up such as `<table>`, `<tr>`, `<th>`, `<td>` etc. to present the content as a data table.
- Mark-up "SUN", "MON",  "TUE" etc. table headers using `<th>` element.

### Actual Behavior
The table content such as "SUN", "MON", "TUE" etc. is presented using `<div>` elements.

### Reproduction Link
<a href="https://codepen.io/apfrod/pen/WNZLRrb" target="_blank">https://codepen.io/apfrod/pen/WNZLRrb</a>

**Comment by KaelWD** [comment]
This should just use ARIA roles instead, actual tables have some pretty annoying layout implications.
